Microsoft/team-explorer-everywhere
# Fix crash when sharing Linux workstation

## Summary

This fixes an issue that was mostly noticed on Linux, with both the TEE CLC or the Eclipse plugin, when more than one user is involved.  Examples:

* [TFS Command Line Client crashes in Linux](https://social.msdn.microsoft.com/Forums/vstudio/en-US/37561345-25d0-4bd9-8d3d-40531c7c117a/tfs-command-line-client-crashes-in-linux?forum=tee)
* [tf command is not running on Centos 7](https://social.msdn.microsoft.com/Forums/vstudio/en-US/cfec6f9b-e0b5-4231-96a1-40fa58167afc/tf-command-is-not-running-on-centos-7?forum=tee)
* [Team foundtaion server plugin eclipse - multiple users on ubuntu](https://social.msdn.microsoft.com/Forums/vstudio/en-US/541f70c1-92b6-4903-9a07-48dcfe186fc2/team-foundtaion-server-plugin-eclipse-multiple-users-on-ubuntu?forum=tee)

## Implementation details

Until pull request Microsoft/ApplicationInsights-Java#279 is merged and Microsoft/ApplicationInsights-Java is released, this pull request introduces the `0.9.5.1` custom version of `applicationinsights-core`, which was built from the [v0.9.5 tag](https://github.com/Microsoft/ApplicationInsights-Java/commit/6ec7a748f229942b3fa0892d5043da23594d3ee8) plus the changes in Microsoft/ApplicationInsights-Java#279.

### How version 0.9.5.1 was built

In the Microsoft/ApplicationInsights-Java clone, using Cygwin:

1. Create the [fix_shared_tmpdir_0.9.5](https://github.com/olivierdagenais/ApplicationInsights-Java/commits/fix_shared_tmpdir_0.9.5) branch:

    ```
    git checkout -b fix_shared_tmpdir_0.9.5 v0.9.5
    ```
2. Cherry-pick the changes from the `fix_shared_tmpdir` branch:

    ```
    git cp f4569e3
    git cp ee94205
    git cp 57f9d82
    git cp 269f78e
    git cp f468c0e
    git cp f57700f
    git cp b52975f
    git cp e16360e
    ```
3. Update version to 0.9.5.1 via commit Microsoft/ApplicationInsights-Java@392d56edd6407d81eb88ea1256a8206c7144fe10
4. Augment release notes via commit Microsoft/ApplicationInsights-Java@06eaba0bd674af14dc0dc14781e4e97eca5ad9aa
5. Configure to use JDK 1.6.0u45:

    ```
    export JAVA_JRE_6=e:/Software/Java/6/jdk1.6.0_45/jre
    ```
6. Rebuild.  The JAR we want will be under `core/build/libs`.

    ```
    ./gradlew clean testAndCollect -DisRelease=true
    ```
7. Because I used Cygwin, the Windows DLLs were not compiled.  What I did, instead, was extract them from the official 0.9.5 release JAR and inserted them into our custom 0.9.5.1 JAR.


## Manual testing

1. Rebuild TEE:

    ```
    cd build
    /cygdrive/c/dev/apache-ant-1.9.6/bin/ant -Ddir.machine.build-runtime=c:/users/olivida/dev/eclipseTargets/352/
    cd ..
    ```
2. Copy the contents of `build/output/bin` to a Fedora 22 VM.
3. Install the TEE Eclipse plugin from `eclipseplugin-update-archive/TFSEclipsePlugin-UpdateSiteArchive-14.0.3.zip` into Eclipse Mars.2 (4.5.2) and restart Eclipse.
4. Browsing the `/tmp` folder, I see a sub-folder was created for the current user (`oli`), under which I can find `AppInsights` and `transmissions` sub-folders (both owned by the current user).
5. SSH to the same VM as another user.
6. This time, unzip `clc/TEE-CLC-14.0.3.zip` into the `/tmp` folder.
7. Run `/tmp/TEE-CLC-14.0.3/tf eula` to display the EULA and accept the terms.
8. Run `/tmp/TEE-CLC-14.0.3/tf workspace`.
9. Browsing the `/tmp` folder: I see a sub-folder was created for the second user, under which I can find `AppInsights` and `transmissions` sub-folders (both owned the second user).
10. There are no `AppInsights` and `transmissions` sub-folders in the `/tmp` folder. 

Mission accomplished!
